src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js

```diff
@@ -261,280 +212,337 @@ class BasePaymentsPage extends React.Component {
+                                addBankAccountRoute={ROUTES.SETTINGS_ADD_BANK_ACCOUNT}
+                                addDebitCardRoute={ROUTES.SETTINGS_ADD_DEBIT_CARD}
+                                popoverPlacement="bottom"
+                            >
+                                {(triggerKYCFlow) => (
+                                    <MenuItem
+                                        title={translate('common.transferBalance')}
+                                        icon={Expensicons.Transfer}
+                                        onPress={triggerKYCFlow}
+                                        shouldShowRightIcon
+                                        disabled={props.network.isOffline}
+                                    />
+                                )}
+                            </KYCWall>
+                        </View>
+                    )}
+                </>
+            )}
+            <Text style={[styles.ph5, styles.textLabelSupporting, styles.mb1]}>{translate('paymentsPage.paymentMethodsTitle')}</Text>
+        </>
+    );
+
+    useEffect(() => {
+        PaymentMethods.openPaymentsPage();
+    }, []);
+
+    useEffect(() => {
+        // If the user was previously offline, skip debouncing showing the loader
+        if (!props.network.isOffline) {
+            updateShouldShowLoadingSpinner();
+        } else {
+            debounceSetShouldShowLoadingSpinner();
+        }
+    }, [props.network.isOffline, debounceSetShouldShowLoadingSpinner, updateShouldShowLoadingSpinner]);
+
+    useEffect(() => {
+        if (props.shouldListenForResize) {
+            setMenuPosition();
+        }
+
+        if (shouldShowDefaultDeleteMenu || showPassword.shouldShowPasswordPrompt) {
+            // We should reset selected payment method state values and close corresponding modals if the selected payment method is deleted
+            let shouldResetPaymentMethodData = false;
+
+            if (paymentMethod.selectedPaymentMethodType === CONST.PAYMENT_METHODS.BANK_ACCOUNT && _.isEmpty(props.bankAccountList[paymentMethod.methodID])) {
+                shouldResetPaymentMethodData = true;
+            } else if (paymentMethod.selectedPaymentMethodType === CONST.PAYMENT_METHODS.DEBIT_CARD && _.isEmpty(props.cardList[paymentMethod.methodID])) {
+                shouldResetPaymentMethodData = true;
+            } else if (paymentMethod.selectedPaymentMethodType === CONST.PAYMENT_METHODS.PAYPAL && _.isEmpty(props.payPalMeData)) {
+                shouldResetPaymentMethodData = true;
+            }
+            if (shouldResetPaymentMethodData) {
+                // Close corresponding selected payment method modals which are open
+                if (shouldShowDefaultDeleteMenu) {
+                    hideDefaultDeleteMenu();
+                } else if (showPassword.shouldShowPasswordPrompt) {
+                    hidePasswordPrompt();
+                }
+            }
+        }
```

Comment:
This can be a separate Effect

Reply:
Hi, @Santhosh-Sellavel 
Can you please check this again?
I think it is not efficient to separate useEffect because of use of `shouldResetPaymentMethodData`

Reply:
Can you explain please? How it's not efficient?

Reply:
In this block, `hideDefaultDeleteMenu ` and `hidePasswordPrompt ` are related with `shouldResetPaymentMethodData` so I think there is no separatable blocks
```
    useEffect(() => {
        if (!shouldShowDefaultDeleteMenu && !showPassword.shouldShowPasswordPrompt) {
            return;
        }

        // We should reset selected payment method state values and close corresponding modals if the selected payment method is deleted
        let shouldResetPaymentMethodData = false;

        if (paymentMethod.selectedPaymentMethodType === CONST.PAYMENT_METHODS.BANK_ACCOUNT && _.isEmpty(props.bankAccountList[paymentMethod.methodID])) {
            shouldResetPaymentMethodData = true;
        } else if (paymentMethod.selectedPaymentMethodType === CONST.PAYMENT_METHODS.DEBIT_CARD && _.isEmpty(props.cardList[paymentMethod.methodID])) {
            shouldResetPaymentMethodData = true;
        } else if (paymentMethod.selectedPaymentMethodType === CONST.PAYMENT_METHODS.PAYPAL && _.isEmpty(props.payPalMeData)) {
            shouldResetPaymentMethodData = true;
        }
        if (shouldResetPaymentMethodData) {
            // Close corresponding selected payment method modals which are open
            if (shouldShowDefaultDeleteMenu) {
                hideDefaultDeleteMenu();
            } else if (showPassword.shouldShowPasswordPrompt) {
                hidePasswordPrompt();
            }
        }
    }, [
        hideDefaultDeleteMenu,
        hidePasswordPrompt,
        paymentMethod.methodID,
        paymentMethod.selectedPaymentMethodType,
        props.bankAccountList,
        props.cardList,
        props.payPalMeData,
        shouldShowDefaultDeleteMenu,
        showPassword.shouldShowPasswordPrompt,
    ]);
```
Can you please let me know which block can be separated more?

Reply:
Yeah, we can do much and I expected the same.